ui-components/src/commonMain/kotlin/org/jetbrains/kotlinconf/ui/components/MainNavigation.kt

```diff
@@ -61,29 +62,30 @@ private fun MainNavigationButton(
     )
 }
 
-data class MainNavDestination<T : Any>(
+data class MainNavDestination(
     val label: String,
     val icon: DrawableResource,
-    val route: T,
+    val route: Any,
+    val routeClass: KClass<*>? = null,
 )
 
 @Composable
-fun <T : Any> MainNavigation(
-    currentDestination: MainNavDestination<T>,
-    destinations: List<MainNavDestination<T>>,
-    onSelected: (MainNavDestination<T>) -> Unit,
+fun MainNavigation(
+    currentDestination: MainNavDestination?,
+    destinations: List<MainNavDestination>,
+    onSelect: (MainNavDestination) -> Unit,
     modifier: Modifier = Modifier,
 ) {
     Row(
         modifier = modifier.fillMaxWidth().padding(horizontal = 4.dp),
         horizontalArrangement = Arrangement.spacedBy(8.dp)
     ) {
-        destinations.forEach {
+        destinations.forEach { destination ->
             MainNavigationButton(
-                iconResource = it.icon,
-                contentDescription = it.label,
-                selected = it == currentDestination,
-                onClick = { onSelected(it) },
+                iconResource = destination.icon,
+                contentDescription = destination.label,
+                selected = destination == currentDestination,
+                onClick = { onSelect(destination) },
```

Comment:
According to the design mockup, the active icon should not only be highlighted, but it is also filled variant instead of outlined.

Reply:
Great catch! Fixed.